### PR TITLE
Add Support for Scikit-Learn.

### PR DESCRIPTION
**Pull request by elibol**

### CODE REVIEW

**Comment by elibol** [comment]
Update to lazy-initialize models.

File: nums/sklearn.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    from sklearn.preprocessing import StandardScaler, RobustScaler
+
+    sklearn_clses = [StandardScaler, RobustScaler]
+    return (build_sklearn_actor(sklearn_cls) for sklearn_cls in sklearn_clses)
+
+
+(StandardScaler, RobustScaler) = build_preprocessors()
+
+
+def exec_parallel(size, features):
```
Comment (dlzou) [inline nums/sklearn.py:165]: This looks pretty similar to code in test_sklearn.py, is it redundant?

File: nums/core/systems/systems.py [hunk 2]
```diff
@@ -59,6 +63,16 @@ def register(self, name: str, func: callable, remote_params: dict = None):
     def call(self, name: str, args, kwargs, device_id: DeviceID, options: Dict):
         return self._remote_functions[name](*args, **kwargs)
 
+    def register_actor(self, name: str, cls: type):
+        assert name not in self._actors
```
Comment (vinamrabenara) [inline nums/core/systems/systems.py:67]: shall we add a error message that the name is not unique?

File: nums/core/systems/systems.py [hunk 3]
```diff
@@ -204,6 +220,24 @@ def num_cores_total(self) -> int:
...
+
+    def make_actor(self, name: str, *args, device_id: DeviceID = None, **kwargs):
+        # Distribute actors round-robin over devices.
+        if device_id is None:
+            device_id = self._devices[self._actor_node_index]
+            self._actor_node_index = (self._actor_node_index + 1) % len(self._devices)
+        actor = self._actors[name]
+        node = self._device_to_node[device_id]
+        node_key = self._node_key(node)
+        options = {"resources": {node_key: 1.0 / 10 ** 4}}
```
Comment (vinamrabenara) [inline nums/core/systems/systems.py:235]: is 10**4 something specific or just a large number to accomodate many actors?

**Review by vinamrabenara — APPROVED** [review]

File: nums/core/systems/systems.py (hunk 3 above)
Comment (elibol) [reply nums/core/systems/systems.py:235]: Just a large number. I think working precision of resources is low (at the time I wrote this), which is why 10^4 is used and not something larger.

File: nums/core/systems/systems.py (hunk 2 above)
Comment (elibol) [reply nums/core/systems/systems.py:68]: This is low-level, we don't expect users to run this. If we make a mistake we want to know about it ;)

**Pull request merged**